terraform/aws/best-practice/aws-s3-object-lock-not-enabled.tf

```diff
@@ -0,0 +1,46 @@
+# pass
+
+resource "aws_s3_bucket" "enabled_via_object" {
+  bucket = "test-bucket"
+  acl    = "private"
+
+  object_lock_configuration = {
+    object_lock_enabled = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket" "enabled_via_block" {
+  bucket = "test-bucket"
+  acl    = "private"
+
+  object_lock_configuration {
+    object_lock_enabled = "Enabled"
+  }
+}
+
+# failure
+# ruleid: aws-s3-object-lock-not-enabled
+resource "aws_s3_bucket" "disabled_via_object" {
+  bucket = "test-bucket"
+  acl    = "private"
+
+  object_lock_configuration = {
+    object_lock_enabled = "Disabled"
+  }
+}
+# ruleid: aws-s3-object-lock-not-enabled
+resource "aws_s3_bucket" "disabled_via_block" {
+  bucket = "test-bucket"
+  acl    = "private"
+
+  object_lock_configuration {
+    object_lock_enabled = "Disabled"
+  }
+}
+
+# unknown
+
+resource "aws_s3_bucket" "default" {
```

Comment:
Do you have a preference for whether this one gets flagged? Currently, the rule does flag here; however, this would alert on many S3 configurations.

If you want to alert here, it'll need a test annotation

Reply:
It may cause false positives (which is valid for all unknown cases). Do you think it's worth increasing coverage for the sake of having more false positives?

Reply:
Same as above